lib/config/configuration.go

```diff
@@ -1935,9 +1938,14 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 
 	// If this process is trying to join a cluster as an application service,
 	// make sure application name and URI are provided.
-	if slices.Contains(splitRoles(clf.Roles), defaults.RoleApp) &&
-		(clf.AppName == "" || clf.AppURI == "") {
-		return trace.BadParameter("application name (--app-name) and URI (--app-uri) flags are both required to join application proxy to the cluster")
+	if slices.Contains(splitRoles(clf.Roles), defaults.RoleApp) {
+		if clf.AppName == "" {
+			return trace.BadParameter("to join application proxy to the cluster application name is required, provide it with --app-name or --name")
+		}
+
+		if clf.AppURI == "" && clf.AppCloud == "" {
+			return trace.BadParameter("to join application proxy to the cluster application URI or Cloud identifier is required, provide it with --app-uri, --uri or --cloud")
+		}
```

Comment:
it looks like `--app-name` and `--app-uri` are deprecated, can we avoid recommending them? https://github.com/gravitational/teleport/blob/50bd4314271bc48c636c0c39470b4fc6e2fe2617/tool/teleport/common/teleport.go#L151

Reply:
I had that thought too, however `Configure()` is reachable from other commands too: https://github.com/gravitational/teleport/blob/9800dc37f6a1eeeb33ace7550581fb2bd3ab8570/tool/teleport/common/teleport.go#L398-L412

In particular, `teleport start` does not support `--name` or `--uri` or `--cloud`.

If we were to omit e.g. `--app-name` the following session would become likely:

```
$ teleport start --roles app
ERROR: to join application proxy to the cluster application name is required, provide it with --name

$ teleport start --roles app --name azure
teleport: error: unknown long flag '--name'
```

Reply:
If all three are listed here, it can become a trial and error too (e.g. `teleport app start --app-uri <uri>`). I think Ideally if we can show different messages for different input commands.

Reply:
> If all three are listed here, it can become a trial and error too (e.g. `teleport app start --app-uri <uri>`). I think Ideally if we can show different messages for different input commands or somehow handle it on the kingpin side.

This is doable, we simply need to pass the name of the command as an argument for the `Configure()` function. I agree that we should avoid confusing users if at all possible.

Reply:
I've added a flag to choose between two sets of flags (`--app-name`, `--app-uri` vs `--name`, `--uri`, `--cloud`). The errors are made precise too, i.e. we only complain about the flags that are actually missing.